### DISCUSSION TITLE
using ts-ls, is there a way to show all errors in a project?

### DISCUSSION THREAD

**Discussion by Askath** [emacs-lsp/lsp-mode#4505] [Q&A] [answered]
Hello, 
I want to check my whole project for possible typescript and eslint errors. The servers are set up however I only get diagnostics for open buffers. 
Is there a way to analyze all files in a project? (or maybe a simple way to open all files in buffers?)

**Comment by yyoncho** [comment, marked as answer]
You can use lsp-treemacs, helm-lsp, lsp-ivy and so on.

**Reply by Askath** [reply]
Those still only show errors in opened buffers or is there a way to check all files without having to open every file?

**Reply by yyoncho** [reply]
These are all errors that the server has sent to us.

**Reply by Askath** [reply]
so the answer is no, gotcha

**Reply by yyoncho** [reply]
Typically the servers send all errors per project. I suspect that eslint server is not doing so.
